### PR TITLE
Contrib docs

### PR DESCRIPTION
**Pull request by bridgetkromhout**
Let's set up the right framework for contributors to succeed.

### CODE REVIEW

**Comment by bridgetkromhout** [comment]
Per @cheftako - for each person's PRs, we will require approval from a different employer than their own.

File: CONTRIBUTOR_LADDER.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    * Occasionally submit PRs
+    * Contribute to the documentation
+    * Show up at meetings, takes notes
+    * Answer questions from other community members
+    * Submit feedback on issues and PRs
+    * Test releases and patches and submit reviews
+    * Run or helps run events
+    * Promote the project in public
+    * Help run the project infrastructure
+    * [TODO: other small contributions]
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
    * Quick starts, recipe book, examples, blogs (including end user blogs)
    * Share use-cases (blog posts, live/recorded webinars...)
    * Share example/reusable compositions [eventually we want store these in artifacthub.io]
```

File: CONTRIBUTOR_LADDER.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+CI/CD automation.  If these rights do not apply to you, then it makes sense to combine this role with the Contributor role.  For most projects, Org Member is the \
+lowest level with clearly defined responsibilities and rights, because it is tied to permissions on the project's code repository. -->
+<!--TODO: project leads to fill in exact details of this role for your project-->
+Description: An Organization Member is an established contributor who regularly participates in the project. Organization Members have privileges in both project repositories and elections, and as such are expected to act in the interests of the whole project.
+
+An Organization Member must meet the responsibilities and has the requirements of a Contributor, plus:
+
+* Responsibilities include:
+    * Continues to contribute regularly, as demonstrated by having at least [TODO: Number] [TODO: Metric] a year, as demonstrated by [TODO: contributor metrics source]. <!-- Example: "as demonstrated by having at least 50 GitHub contributions per year, as shown by Devstats"-->
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
    * Regularly contributes to code, documentation, issues, and discussions. <!-- as demonstrated by having at least [TODO: Number] [TODO: Metric] a year, as demonstrated by [TODO: contributor metrics source].  Example: "as demonstrated by having at least 50 GitHub contributions per year, as shown by Devstats"-->
```

File: CONTRIBUTOR_LADDER.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+* Requirements:
+    * Must have successful contributions to the project, including at least one of the following:
+        * [TODO: Number] accepted PRs,
+        * Reviewed [TODO: Number] PRs,
+        * Resolved and closed [TODO: Number] Issues,
+        * Become responsible for a key project management area,
+        * Or some equivalent combination or contribution
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md:93]:
```suggestion
<!--TODO
    * Must have successful contributions to the project, including at least one of the following:
        * [TODO: Number] accepted PRs,
        * Reviewed [TODO: Number] PRs,
        * Resolved and closed [TODO: Number] Issues,
        * Become responsible for a key project management area,
        * Or some equivalent combination or contribution
 -->
```

File: CONTRIBUTOR_LADDER.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    * Must have been contributing for at least [TODO: Number] months
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
    * Must have been contributing for at least 3 months
```

File: CONTRIBUTOR_LADDER.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    * Must be actively contributing to at least one project area
+    * Must have two sponsors who are also Organization Members, at least one of whom does not work for the same employer
+    * [TODO: other requirements] <!--TODO: such as repository access or enabling 2FA on their GitHub account -->
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
    * Must have 2FA enabled on their GitHub account.
```

File: CONTRIBUTOR_LADDER.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+
+* Privileges:
+    * May be assigned Issues and Reviews
+    * May give commands to CI/CD automation
+    * Entitled to vote in the [TODO: appropriate election]
+    * Can be added to [TODO: Repo Host] teams
+    * Can recommend other contributors to become Org Members
+    * [TODO: Other Privileges]
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
* Privileges:
    * May be assigned Issues and Reviews
    * May give commands to CI/CD automation
    * Can sponsor other contributors to become Org Members
```

File: CONTRIBUTOR_LADDER.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+
+The process for a Contributor to become an Organization Member is as follows:
+
+<!-- the process of becoming an organization member is going to depend strongly on how your project manages its infrastructure. TODO: Project leads to fill in exact details of how a contributor becomes an organization member-->
+1.
+2.
+3.
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
```

File: CONTRIBUTOR_LADDER.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+
+### Reviewer
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md:112]: To review later this year/early next year.

File: CONTRIBUTOR_LADDER.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+    * Experience as a Contributor for at least [TODO: Number] months
+    * Is an Organization Member
+    * Has reviewed, or helped review, at least [TODO: Number] Pull Requests
+    * Has analyzed and resolved test failures in their specific area
+    * Has demonstrated an in-depth knowledge of the specific area
+    * Commits to being responsible for that specific area
+    * Is supportive of new and occasional contributors and helps get useful PRs in shape to commit
+* Additional privileges:
+    * Has GitHub or CI/CD rights to approve pull requests in specific directories
+    * Can recommend and review other contributors to become Reviewers
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
* Responsibilities include:
    * TBD
* Requirements:
    * TBD
* Additional privileges:
    * TBD
```

File: CONTRIBUTOR_LADDER.md [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+    * Experience as a Reviewer for at least [TODO: Number] months
+    * Demonstrates a broad knowledge of the project across multiple areas
+    * Is able to exercise judgment for the good of the project, independent of their employer, friends, or team
+    * Mentors other contributors
+    * Can commit to spending at least [TODO: Number] hours per month working on the project
+* Additional privileges:
+    * Approve PRs to any area of the project
+    * Represent the project in public as a Maintainer
+    * Communicate with the CNCF on behalf of the project
+    * Have a vote in Maintainer decision-making meetings
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
* Responsibilities include:
    * TBD
* Requirements:
    * TBD
* Additional privileges:
    * TBD
```

File: CONTRIBUTOR_LADDER.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+    * Has GitHub or CI/CD rights to approve pull requests in specific directories
+    * Can recommend and review other contributors to become Reviewers
+    
+<!-- TODO: define how this works with your specific system.  For example:  "Is listed as Approver in the OWNERS file for certain directories. -->
+
+
+The process of becoming a Reviewer is:
+<!-- TODO: define your exact process here.  What's below is given as an example process for a project that uses Owners files and has defined teams for each project area -->
+1. The contributor is nominated by opening a PR against the appropriate repository, which adds their GitHub username to the OWNERS file for one or more directories.
+2. At least two members of the team that owns that repository or main directory, who are already Approvers, approve the PR.
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
```

File: CONTRIBUTOR_LADDER.md [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+    * Represent the project in public as a Maintainer
+    * Communicate with the CNCF on behalf of the project
+    * Have a vote in Maintainer decision-making meetings
+    
+
+Process of becoming a maintainer:
+<!-- TODO: this process will vary widely across projects, both because of project code structure, and because of project governance.  For example, in some projects the Steering Committee approves new Maintainers.  What's below is just an example from a simple project in which the maintainers are also the project leaders, and which uses GitHub OWNERS files. -->
+1. Any current Maintainer may nominate a current Reviewer to become a new Maintainer, by opening a PR against the root of the [TODO: main repository name] adding the nominee as an Approver in the OWNERS file.
+2. The nominee will add a comment to the PR testifying that they agree to all requirements of becoming a Maintainer.
+3. A majority of the current Maintainers must then approve the PR.
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md]:
```suggestion
```

File: CONTRIBUTOR_LADDER.md [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+Involuntary removal or demotion is handled through a vote by a majority of the current Maintainers.
+
+## Stepping Down/Emeritus Process
+If and when contributors' commitment levels change, contributors can consider stepping down (moving down the contributor ladder) vs moving to emeritus status (completely stepping away from the project).
+
+Contact the Maintainers about changing to Emeritus status, or reducing your contributor level.
+
+## Contact
+* For inquiries, please reach out to:
+    *  <!-- TODO: fill in contact info for appropriate group or person for contributor mentorship-->
```
Comment (a-hilaly) [inline CONTRIBUTOR_LADDER.md:190]:
```suggestion
    *  <!-- TODO: fill in contact info for appropriate group or person for contributor mentorship-->
    *  <!-- TODO: Create a slack channel for CNC -->
```

**Review by a-hilaly — APPROVED** [review]

**Comment by bridgetkromhout** [comment]
Discussed in meeting with @cheftako @justinsb @a-hilaly

**Pull request merged**